File: system-test/java/src/test/java/com/optivem/eshop/systemtest/legacy/mod03/smoke/external/ErpSmokeTest.java
package com.optivem.eshop.systemtest.legacy.mod03.smoke.external;

import com.optivem.eshop.systemtest.legacy.mod03.base.BaseRawTest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.net.URI;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ErpSmokeTest extends BaseRawTest {
    private static final String HEALTH_ENDPOINT = "/health";

    @BeforeEach
    void setUp() {
        setUpExternalHttpClients();
    }

    @Test
    void shouldBeAbleToGoToErp() throws Exception {
        var uri = URI.create(getErpBaseUrl() + HEALTH_ENDPOINT);
        var request = HttpRequest.newBuilder()
                .uri(uri)
                .GET()
                .build();

        var response = erpHttpClient.send(request, HttpResponse.BodyHandlers.ofString());

        assertEquals(200, response.statusCode());
    }
}

File: system-test/dotnet/SystemTests/Legacy/Mod02/SmokeTests/External/ErpSmokeTest.cs
using SystemTests.Legacy.Mod02.Base;
using Xunit;

namespace SystemTests.Legacy.Mod02.SmokeTests.External;

public class ErpSmokeTest : BaseRawTest
{
    private const string HealthEndpoint = "/health";

    public override async Task InitializeAsync()
    {
        await base.InitializeAsync();
        SetUpExternalHttpClients();
    }

    [Fact]
    public async Task ShouldBeAbleToGoToErp()
    {
        var uri = new Uri(_configuration.ErpBaseUrl + HealthEndpoint);
        var request = new HttpRequestMessage(HttpMethod.Get, uri);

        var response = await _erpHttpClient!.SendAsync(request);

        Assert.Equal(200, (int)response.StatusCode);
    }
}











